tests/utils/test_split_large_files.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pathlib
+    from collections.abc import Callable
+
+
+import numpy as np
+import pandas as pd
+import pyarrow.parquet as pq
+import pytest
+
+from nemo_curator.utils.split_large_files import parse_args, split_parquet_file_by_size
+
+
+@pytest.fixture
+def parquet_file_factory(tmp_path: pathlib.Path):
+    def _(num_row_groups: int = 1) -> pathlib.Path:
+        num_rows = 500_000
+        rng = np.random.default_rng(seed=2)
+        df = pd.DataFrame(
+            {
+                "id": np.arange(num_rows),
+                "value1": rng.random(num_rows),
+                "value2": rng.integers(0, 1000, num_rows),
+                "category": rng.choice(["A", "B", "C", "D"], num_rows),
+                "timestamp": pd.to_datetime("2023-01-01") + pd.to_timedelta(np.arange(num_rows), unit="s"),
+            }
+        )
```

Comment:
Can you add a comment about how much this data is in memory? And to confirm we split by uncompressed file size or file size? I believe the former